src/_support/Step/Acceptance/Administrator/MenuItemStep.php

```diff
@@ -0,0 +1,408 @@
+ * @subpackage  AcceptanceTester.Step
+ *
+ * @copyright   Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Step\Acceptance\Administrator;
+
+use Page\Acceptance\Administrator;
+
+/**
+ * Acceptance Step object class contains suits for MenuItem.
+ *
+ * @package  Step\Acceptance\Administrator
+ *
+ * @since    4.0.0
+ */
+class MenuItemStep extends Admin
+{
+	/**
+	 * Create A New MenuItem
+	 *
+	 * @param   \AcceptanceTester $I                 The Acceptance Tester
+	 * @param   string            $menuItemName      The Title Of Menu Item
+	 * @param   string            $menuItemAlias     The Alias Of Menu Item
+	 * @param   string            $title             The Title Of Category or Article or Contact or News Feed
+	 * @param   string            $menuItemType      The type of MenuItem
+	 * @param   string            $optionForMenuType ( Archived Articles, Featured, Category Blog, etc)
+	 * @param   string            $tag               Tag
+	 *
+	 * @since   4.0.0
+	 * @return void
+	 */
+	public function menuItem(\AcceptanceTester $I ,$menuItemName, $menuItemAlias, $title, $menuItemType = 'Articles', $optionForMenuType = 'List All Categories', $tag = 'joomlaorg' )
+	{
+		$I->comment('I am going to create a menu item');
+		$I->doAdministratorLogin();
+		$I->amOnPage(Administrator\MenuItemListPage::$url);
+		$I->checkForPhpNoticesOrWarnings();
+		$I->waitForText(Administrator\MenuItemFormPage::$pageTitleText);
+		/**
+		 * Creating A New Menu item
+		 *  1. click on "new" botton
+		 *  2. fill the Menu Select type
+		 *  3. fill two fields : menu item name and alias
+		 *  4. click on "save" button
+		 */
+		$I->click(['id' => "menu-collapse"]);
+		$I->clickToolbarButton('new');
+		$I->fillField(Administrator\MenuItemFormPage::$menuItemTitle, $menuItemName);
+		$I->fillField(Administrator\MenuItemFormPage::$menuItemAlias, $menuItemAlias);
+
+		// Select option from dropdown menu  : MAIN MENU
+		$I->click(Administrator\MenuItemFormPage::$menuDropDown);
+		$option = $I->grabTextFrom(Administrator\MenuItemFormPage::$selectOption);
+		$I->selectOption(Administrator\MenuItemFormPage::$menuDropDown, $option);
+		/**
+		 * Menu Type
+		 * Articles
+		 * News Feed
+		 * Search
+		 * Smart Search
+		 * Configuration Manager
+		 * Contacts
+		 * System Links
+		 * Tags
+		 * Wrapper
+		 */
+		$I->waitForText('Select');
+		$I->click('Select');
+		$I->waitForElement(Administrator\MenuItemFormPage::$menuTypeModal, TIMEOUT);
+		$I->switchToIFrame("Menu Item Type");
+		switch ($menuItemType)
+		{
+			case 'Articles' :
+				$I->click(['link' => 'Articles']);
+				/**
+				 * Options under Articles
+				 * Archived Article  .
+				 * Featured Article  .
+				 * Create Article  .
+				 * Category Blog  .
+				 * Category List  .
+				 * List All Categories .
+				 * Single Article .
+				 */
+				switch ($optionForMenuType) {
```

Comment:
well to be honest about this, I think you can improve this approach, instead of having a big switch, why not write a function that takes the menu item type as a parameter and performs actions accordingly?

Reply:
I tried that but some options involve IFrame, or select options textbox. There are lot of  varieties.